release/entrypoint.sh

```diff
@@ -0,0 +1,137 @@
+echo "Project: $PROJECT"
+
+#---------------#
+# Parse Appdata #
+#---------------#
+
+# get appdata filename:
+APPDATA="$(basename "$(find data -name "*appdata*")")"
+
+# get the version and release notes from the latest release entry in the appdata
+VERSION="$(xmlstarlet sel -t -v '//release[1]//@version' -n data/"$APPDATA")"
+echo "Version: $VERSION"
+
+# get the release notes, remove any empty lines & padded spacing
+RELEASE_NOTE_RAW="$(xmlstarlet sel -t -v '//release[1]' -n data/"$APPDATA"| awk 'NF'| awk '{$1=$1}1')"
+echo "Release Note Content:"
+echo -e "$RELEASE_NOTE_RAW\n"
+
+#-----------------------#
+# Create Github Release #
+#-----------------------#
+
+# add ul stars to the notes and replace any newlines with a newline character instead
+MARKDOWN_NOTES="$(echo "$RELEASE_NOTE_RAW" | awk '$0="* "$0' | awk '{printf "%s\\n", $0}')"
+# add the project name and version to release note
+GITHUB_RELEASE_NOTE="$PROJECT $VERSION is out! \n\nChanges:\n\n$MARKDOWN_NOTES"
+DATA="
+{
+  \"tag_name\": \"$VERSION\",
+  \"target_commitish\": \"$GITHUB_SHA\",
+  \"name\": \"$PROJECT $VERSION Released\",
+  \"body\": \"$GITHUB_RELEASE_NOTE\",
+  \"draft\": false,
+  \"prerelease\": false
+}
+"
+
+# push the release content to github!
+if ! curl --data "$DATA" https://api.github.com/repos/"$GITHUB_REPOSITORY"/releases?access_token="$GITHUB_TOKEN"; then
+  echo "\033[0;31mERROR: Unable to post github release tag information!\033[0m"  && exit 1
+fi
+echo -e "\n\033[1;32mA new github release tag has been created!\033[0m\n"
+
+#-------------------------#
+# Update Debian Changelog #
+#-------------------------#
+
+# make sure we are in sync with HEAD
+git reset --hard HEAD
+
+# get all commit subjects since last tag
+COMMITS="$(git log "$(git describe --tags "$(git rev-list --tags --max-count=1)")"..HEAD --pretty="format:%s")"
+# filter out commits involving translations and commits that don't have a related merge number
+FILTERED_COMMITS="$(echo "$COMMITS" | awk '!/Weblate/' | awk '!/weblate/' | grep '(#')"
+
+# move to the debian packaging branch
+if ! git checkout deb-packaging; then
+  echo "\033[0;31mERROR: Unable to checkout the 'deb-packaging' branch. Does it exist?\033[0m" && exit 1
+fi
+
+if [ -z "$FILTERED_COMMITS" ]; then
+  echo "no changes since last tag, an empty debian changelog will be created."
+  dch -Mv "$VERSION" "no changes since last release."
+else
+  # Get the first changelog entry
+  FIRST_CHANGE="$(echo "$FILTERED_COMMITS" | head -n 1)"
+  # Create a versioned release and add the first line of the changelog
+  dch -Mv "$VERSION" "$FIRST_CHANGE"
+  # iterate over any other changelog entries, if there are any
+  REMAINING_CHANGES="$(echo "$FILTERED_COMMITS"| tail -n +2)"
+  if [ -n "$REMAINING_CHANGES" ]; then
+    while read -r line; do
+      # Append another list item to the changelog
+      dch -Ma "$line"
+    done <<< "$REMAINING_CHANGES"
+  fi
+fi
+
+# Set the release channel/distro
+dch -Mr bionic
```

Comment:
It looks like you're doing the right thing here, but the output we're getting is `unstable`: https://github.com/kgrubb/helloworld/blob/deb-packaging/debian/changelog#L1

I don't see anything in the actions output that looks like an error here either. Hrm

Reply:
For the distribution name, I think that's showing up because I didn't specify a release when I originally created the debian changelogs in my helloworld project. Running the same thing in `camera`, for example, keeps bionic:

![Screenshot from 2019-10-21 15-13-34](https://user-images.githubusercontent.com/8471701/67235708-a585a300-f415-11e9-92ca-9b93aaabf368.png)

Reply:
dch man pages say this for `--release` or `-r`:

> Finalize  the  changelog for a release.  Update the changelog timestamp. If the distribution is set to UNRELEASED, change it to the distribution from the previous changelog entry (or another distribution as specified by --distribution).  If there are no
              previous changelog entries and an explicit distribution has not been specified, unstable will be used.

When creating the deb-packaging branch, I didn't specify a distribution (oops :speak_no_evil: ), so I think that's what's causing the issue.

Should we force distribution?

`--force-distribution`
> Force the provided distribution to be used, even if it doesn't match the list of known distributions (e.g. for unofficial distributions).

Reply:
We can probably come back and add a distro option later if we feel like it's really necessary